volatility3/framework/symbols/linux/__init__.py

```diff
@@ -29,6 +29,7 @@ def __init__(self, *args, **kwargs) -> None:
         self.set_type_class('files_struct', extensions.files_struct)
         self.set_type_class('vfsmount', extensions.vfsmount)
         self.set_type_class('kobject', extensions.kobject)
+        self.set_type_class('mnt_namespace', extensions.mnt_namespace)
```

Comment:
Changes to core framework files are.. delicate.  We can do this, but it might be nice to give people an example of how to keep it all self contained.  This would basically mean calling this on the `symbol_table` of the `kernel` within the plugin itself.  This is just because not every plugin will be able to make core changes like this, so the more examples of how to do it without specialist help, the better.

Reply:
Ouch haha... just wait for the next pull request :). 
I really love the flexibility this feature gives to the framework. Being able to extend types allows to make the differences between kernel versions transparent to the plugins. That's fantastic and super powerful.
We can move these changes to the plugin, however I think moving this kind of changes from the core to the surface leads to have duplicate code in the future. I believe it is better these differences be managed by the framework and not by the plugins. Anyway, I understand you being reluctant to these sort of changes. If you still believe it's not appropriate just let me know or again feel free to move that code.

Reply:
Ok, that's fine.  I've seen the next pull request and I agree with you that it should probably live somewhere central, but that means it'll need review by our core linux guy, @atcuno, to make sure he's happy with it.

Reply:
Cool. I moved one more function here which made everything easier from the plugin side. That allowed me to remove all that dodgy stuff we were discussing above in the first comments. So, it feels like it was the place to be.